metis-common/metis-common-mongo/src/main/java/eu/europeana/metis/mongo/connection/MongoProperties.java

```diff
@@ -78,14 +78,19 @@ public void setAllProperties(String[] hosts, int[] ports, String authenticationD
    * @param password The password. Can be null, in which case no authentication takes place.
    * @param enableSsl Whether to enable SSL connections.
    * @param readPreferenceValue The read preference. Can be null, where then the default applies
+   * @param applicationName
    * @throws E In case either of the arrays is null, or their lengths don't match.
    */
   public void setAllProperties(String[] hosts, int[] ports, String authenticationDatabase,
-      String username, String password, boolean enableSsl, ReadPreferenceValue readPreferenceValue)
+      String username, String password, boolean enableSsl, ReadPreferenceValue readPreferenceValue,
+      String applicationName)
```

Comment:
We need to be  cautious here, this is a library used by others.
I think it there might be a conflict in corelib at least.
It would be better to create another method and keep the already existent that calls that "new" method with null as the applicationName to get the default.
We could even deprecate the old method if we want to, but we should have follow up tickets to update the corresponding projects

Reply:
That's a good catch, I was not aware.
I made a "new" method of `setAllProperties` without the `applicationName` input variable. I also added the `@Deprecated` annotation to it and set the default value of application name as empty.